crates/samples/echomain/src/test.rs

```diff
@@ -67,14 +135,63 @@ async fn test_fabric_client() {
     );
     assert_ne!(stateless_replica.node_name, HSTRING::new());
 
-    // Restart the stateless instance by removing it.
     let mgmt = fc.get_service_manager();
-    let desc = RemoveReplicaDescription {
-        node_name: stateless_replica.node_name,
-        partition_id: single.id,
-        replica_or_instance_id: stateless_replica.instance_id,
+    // register service notification filter
+    let filter_handle = {
+        let desc = ServiceNotificationFilterDescription {
+            name: service_uri.clone(),
+            flags: ServiceNotificationFilterFlags::NamePrefix,
+        };
+        // register takes more than 1 sec.
+        mgmt.register_service_notification_filter(&desc, Duration::from_secs(10))
```

Comment:
So basically , the caller need to retry if they don't get expected count ?? Notification will eventually catchup on that .. Any possibility that notification can be lost .. Is it best effort?

Reply:
Caller needs to retry if the addr is not expected, including the number of addrs are not correct.
AFAIK, new addrs should eventually show up. I will keep looking if there are better ways, since I observed some slowness.